### ISSUE TITLE
NUnit2010 - do not consider Equals call if it doesn't override Object.Equals

### ISSUE THREAD

**Issue by matode** [nunit/nunit.analyzers#728]
If there is a class Foo, implementing Equals in this way:

 ```csharp
class Foo
{
    public bool Equals(Foo objectToCompare
    {
      bool areEqual;
      //some logic modifying areEqual;
      return areEqual;
    }
}
 ```

and Assert like:
 ```csharp
Foo expectedValue = new Foo();
Foo testeeResult = new Foo(); //for simplicity. Some calculation happened instead.
Assert.That(expectedValue.Equals(testeeResult));
 ```

NUnit2010 fix changes it to 
 ```csharp
Assert.That(expectedValue, Is.EqualTo(testeeResult));
 ```

But this test fails with error message:
Expected: Foo
But was: Foo

It would work, if Foo would implement public override bool Equals(object objectToCompare) instead. But this change is not possible for legacy code reason.

It would probably better, if the Analyzer would check, if the called Equals method is an override from object or just an isolated implementation and ignore it in such case.

**Comment by manfred-brands** [comment]
Agreed. If `Equals` is not `Equals(object)` and not `IEquality<Foo>` then it should not raise NUnit2010.

I guess there is a legacy reason `Foo` doesn't derive from `IEquality<Foo>`  but does implement that method.

**Comment by manfred-brands** [comment]
I added a PR to exclude `Equals` implementation if the class doesn't implement the corresponding `IEquality` interface.
However, I'm not sure if this actually should be fixed in `NUnit` itself instead.
If a class A implements `bool Equals(B)` then `Assert.That(B, Is.EqualTo(A))` should pass.
